pkg/engine2/operational_eval/vertex_resource_rule_test.go

```diff
@@ -0,0 +1,220 @@
+package operational_eval
+
+import (
+	"fmt"
+	"testing"
+
+	construct "github.com/klothoplatform/klotho/pkg/construct2"
+	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base2"
+	"github.com/stretchr/testify/assert"
+	gomock "go.uber.org/mock/gomock"
+)
+
+func Test_resourceRuleVertex_Key(t *testing.T) {
+	tests := []struct {
+		name string
+		v    resourceRuleVertex
+		want Key
+	}{
+		{
+			name: "resource rule vertex",
+			v: resourceRuleVertex{
+				Resource: construct.ResourceId{Name: "test"},
+				Rule: knowledgebase.AdditionalRule{
+					If: "test",
+				},
+			},
+			want: Key{
+				Ref: construct.PropertyRef{
+					Resource: construct.ResourceId{Name: "test"},
+				},
+				RuleHash: "9a510ea0226e156085c51625d22ac3679eb7530fd36ba925fcbe211da9c1b373",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+			got := tt.v.Key()
+			assert.Equal(tt.want, got)
+		})
+	}
+}
+
+func Test_resourceRuleVertex_Dependencies(t *testing.T) {
```

Comment:
Side comment, nonblocking:

This test kinda demonstrates why I don't really like structural testing versus behavioural. Reading this test doesn't give any insight into how a resourceRule operates. There's no realistic example rules with expected output. The test is just for testing's sake, and doesn't assist in understanding (which I use in concert with documentation and reading the actual code).

Reply:
i disagree, it assists in the sense of, we offload the determination of the dependencies by feeding the operational rule into the dependency capturer. If you want to know how the dependency capturer works, then thats a different story than this method